discretisedfield/field.py

```diff
@@ -3475,3 +3476,115 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+
+        units : str, optional
+
+            String to set units of the field ``DataArray``.
+
+        Returns
+        -------
+        xarray.DataArray
+
+            Field values DataArray.
+
+        Raises
+        ------
+        TypeError
+
+            If either ``name`` or ``units`` argument is not a string.
+
+        Examples
+        --------
+        >>> import discretisedfield as df
+        ...
+        >>> p1 = (0, 0, 0)
+        >>> p2 = (10, 10, 10)
+        >>> cell = (1, 1, 1)
+        >>> mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+        >>> field = df.Field(mesh=mesh, dim=3, value=(1, 0, 0), norm=1.)
+        ...
+        >>> field
+        Field(...)
+        >>> field.to_xarray()
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10, comp: 3)>
+        ...
+        Coordinates:
+          * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * comp     (comp) <U1 'x' 'y' 'z'
+        >>> xa = field.to_xarray()
+        >>> xa.sel(comp='x')
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10)>
+        ...
+        Coordinates:
+          * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+          * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+            comp     <U1 'x'
+
+        """
+        if not isinstance(name, str):
+            msg = "Name argument must be a string."
+            raise TypeError(msg)
+
+        if units is not None and not isinstance(units, str):
+            msg = "Units argument must be a string."
+            raise TypeError(msg)
+
+        axes = ['x', 'y', 'z']
+
+        data_array_coords = {
+            axis: np.fromiter(self.mesh.axis_points(axis), dtype=float)
+            for axis in axes
+        }
+
+        geo_units_dict = None
+        if 'unit' in self.mesh.attributes:
+            geo_units_dict = {
+                axis: self.mesh.attributes['unit'] for axis in axes
+            }
+
+        if self.dim != 1:
+            data_array_dims = axes + ['comp']
+            if self.components is not None:
+                data_array_coords['comp'] = self.components
+            field_array = self.array
+        else:
+            data_array_dims = axes
+            field_array = np.squeeze(self.array, axis=-1)
+
+        data_array = xr.DataArray(field_array,
+                                  dims=data_array_dims,
+                                  coords=data_array_coords)
+
+        if geo_units_dict is not None:
+            for dim in geo_units_dict:
+                data_array[dim].attrs['units'] = geo_units_dict[dim]
+
+        if units is not None:
+            data_array.attrs['units'] = units
+
+        data_array.name = name
+
+        return data_array
```

Comment:
It might be (structurally) cleaner if we create DataArray at this point and pass all necessary arguments at once (such as `attr`, `name`).

Reply:
I have removed the `data_array.name = name`, and instead, passed it directly like `xr.DataArray(..., name=name)`. Also, I have removed the condition that checks if `units` argument is `None`. I pass it anyway like `xr.DataArray(..., attrs=dict(units=units))`. I think having `units` attribute as None is harmless. 
```Python
for dim in geo_units_dict:
            data_array[dim].attrs['units'] = geo_units_dict[dim]
```
However, I am not able to figure out how to set the `units` attribute for axes before initializing the DataArray. Let me know if you have any ideas.

Reply:
I think the only option would be to first create `DataArrays` for `coords` (where we could pass `attrs`). But I wouldn't suggest doing this as it will complicate the code rather than simplifying it.

Reply:
Exactly what I was thinking. It looks like there is no simple way of doing it.